File: products/admin-desktop/g5-admin/src/api/client/ssh-terminal-bridge.ts
import type { SshTerminalBridgeConnectInput } from "../../types/SshTerminalBridgeConnectInput";
import type { SshTerminalBridgeConnectionResponse } from "../../types/SshTerminalBridgeConnectionResponse";
import { invokeCommand } from "./core";

export async function connectSshTerminalBridge(
  input: SshTerminalBridgeConnectInput,
): Promise<SshTerminalBridgeConnectionResponse> {
  return invokeCommand<SshTerminalBridgeConnectionResponse>(
    "cmd_ssh_terminal_bridge_connect",
    { input },
  );
}

File: g5-admin/src-tauri/src/commands/site/ssh_terminal_bridge.rs
use super::shared::site_command_error;
use crate::app_state::AppState;
use crate::error::CommandResult;
use crate::request_id::next_request_id;
use g5_admin_models::models::ssh::{
    SshTerminalBridgeConnectInput, SshTerminalBridgeConnectionResponse,
};
use tauri::State;

#[tauri::command]
pub async fn cmd_ssh_terminal_bridge_connect(
    state: State<'_, AppState>,
    input: SshTerminalBridgeConnectInput,
) -> CommandResult<SshTerminalBridgeConnectionResponse> {
    let request_id = next_request_id();
    state
        .ssh_terminal_bridge_service()
        .connect(&request_id, input)
        .await
        .map_err(|error| {
            site_command_error(
                "cmd_ssh_terminal_bridge_connect",
                "local-ssh-shell",
                &request_id,
                error,
            )
        })
}
